.github/workflows/ci.yml

```diff
@@ -6,14 +6,94 @@ jobs:
   build-ubuntu:
     strategy:
       matrix:
-        platform: [ubuntu-latest, ubuntu-16.04]
+        platform: [ubuntu-latest,ubuntu-16.04]
     runs-on: ${{ matrix.platform }}
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
     - name: Install dependencies
-      run: sudo apt-get install autoconf automake pkg-config libevent-dev libpcre3-dev libssl-dev
+      run: |
+        sudo apt-get -qq update
+        sudo apt-get install autoconf automake pkg-config libevent-dev libpcre3-dev libssl-dev
     - name: Build
       run: autoreconf -ivf && ./configure && make
+    - name: Setup Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: '3.6'
+
+    - name: Cache pip
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip # This path is specific to Ubuntu
+        # Look to see if there is a cache hit for the corresponding requirements file
+        key: ${{ runner.os }}-pip-${{ hashFiles('tests/test_requirements.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
+          ${{ runner.os }}-
+    - name: Install Python dependencies
+      run: pip install -r tests/test_requirements.txt
+
+    - name: Install Redis Server test dependencies
+      run: |
+        git clone git://github.com/antirez/redis.git --branch unstable
+        cd redis
+        make BUILD_TLS=yes
+        ./utils/gen-test-certs.sh
+        ./src/redis-server --version
+        cd ..
```

Comment:
I think it would be a good idea to cache also the redis-server binary and the test cert files.

Reply:
will do so :+1: